tests/src/serverTestSetError.cpp

```diff
@@ -0,0 +1,50 @@
+#define BOOST_TEST_MODULE serverTestSetError
+
+#include <boost/test/included/unit_test.hpp>
+
+#include <ChimeraTK/ControlSystemAdapter/Testing/ReferenceTestApplication.h>
+
+extern const char* object_name;
+#include <doocs-server-test-helper/ThreadedDoocsServer.h>
+#include <doocs-server-test-helper/doocsServerTestHelper.h>
+
+#include <eq_fct.h>
+
+#include "DoocsAdapter.h"
+#include "serverBasedTestTools.h"
+
+using namespace boost::unit_test_framework;
+using namespace ChimeraTK;
+
+DOOCS_ADAPTER_DEFAULT_FIXTURE_STATIC_APPLICATION
+
+std::string lastErrString = "(unset)";
+static eq_errors checkLocationError(const std::string& property) {
+  auto location = getLocationFromPropertyAddress(property);
+  location->lock();
+  int code = location->get_error();
+  lastErrString = location->get_errorstr();
+  location->unlock();
+
+  return static_cast<eq_errors>(code);
+}
+
+// basic check for error code handling, without associated message source
+BOOST_AUTO_TEST_CASE(testErrorNoMessageSource) {
+  auto check = std::bind(checkLocationError, "//INT/FROM_DEVICE_SCALAR");
```

Comment:
Modern C++ would use a lambda instead of std::bind...

Reply:
lambdas are harder for the debugger. Otherwise, I don't mind the change.